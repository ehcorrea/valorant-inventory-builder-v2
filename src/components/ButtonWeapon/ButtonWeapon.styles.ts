import { TouchableOpacity, View } from 'react-native';
import styled, { css, DefaultTheme } from 'styled-components/native';
import { Image, ImageProps } from 'expo-image';

const buttonModifiers = {
  weapon: (theme: DefaultTheme) => css`
    align-items: center;
    border: solid 2px ${theme.colors.red};
    justify-content: center;
    margin: 5px;
  `,
  'weapon-skin': () => css``,
};

const containerImageModifiers = {
  weapon: () => css``,
  'weapon-skin': () => css`
    height: 80%;
    width: unset;
  `,
};

type ModifierProps = {
  variant: 'weapon' | 'weapon-skin';
};

export const Button = styled(TouchableOpacity)<ModifierProps>`
  ${({ theme, variant }) => css`
    aspect-ratio: 4 / 2;
    background-color: #ffffff14;
    border-radius: ${theme.border.radius.medium}px;
    max-height: ${theme.spacings.rhvalue(100)}px;
    overflow: hidden;
    width: 100%;

    ${buttonModifiers[variant](theme)};
  `}
`;

export const ImageWeapon = styled(Image).attrs<ImageProps>({
  contentFit: 'contain',
})`
  height: 100%;
`;

export const ContainerImage = styled(View)<ModifierProps>`
  ${({ theme, variant }) => css`
    display: flex;
    height: 100%;
    justify-content: center;
    padding: ${theme.spacings.xxxsmall}px;
    width: 100%;

    ${containerImageModifiers[variant]()}
  `}
`;

export const LabelWeapon = styled(View)`
  ${({ theme }) => css`
    align-items: center;
    background-color: ${theme.colors.red};
    height: 20%;
    padding-horizontal: ${theme.spacings.small}px;
    width: 100%;
  `}
`;
