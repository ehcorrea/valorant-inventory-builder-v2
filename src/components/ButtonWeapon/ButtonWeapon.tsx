import { TouchableOpacityProps } from 'react-native';

import * as S from './ButtonWeapon.styles';

export type ButtonWeaponProps = {
  image: string;
} & TouchableOpacityProps;

export function ButtonWeapon({ image, ...props }: ButtonWeaponProps) {
  return (
    <S.Button {...props}>
      <S.ImageWeapon source={image} />
    </S.Button>
  );
}
