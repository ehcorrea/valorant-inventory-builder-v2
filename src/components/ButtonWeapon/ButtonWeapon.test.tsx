import { fireEvent, render } from '@/test/utils';

import { ButtonWeapon, ButtonWeaponProps } from './ButtonWeapon';

const mockedOnPress = jest.fn();

const setup = (props: Partial<ButtonWeaponProps>) => {
  return render(
    <ButtonWeapon
      {...(props as ButtonWeaponProps)}
      onPress={mockedOnPress}
      image="placeholder-image.png"
      testID="weapon-button"
      displayName="Weapon Display Name"
    />
  );
};

describe('given that the <ButtonWeapon> is rendered', () => {
  describe("and is 'weapon' variant", () => {
    test('pressing calls a callback', () => {
      const container = setup({ variant: 'weapon' });
      const button = container.getByTestId('weapon-button');
      fireEvent.press(button);
      expect(mockedOnPress).toHaveBeenCalled();
    });
    test('render image and dont render label', () => {
      const container = setup({ variant: 'weapon' });
      const image = container.getByLabelText('Weapon Display Name button');
      expect(image.props.source).toBe('placeholder-image.png');
      expect(container.queryByText('Weapon Display Name')).toBeFalsy();
      expect(container).toMatchSnapshot();
    });
  });
  describe("and is 'weapon-skin' variant ", () => {
    test('render label', () => {
      const container = setup({
        variant: 'weapon-skin',
      });
      expect(container.getByText('Weapon Display Name')).toBeTruthy();
      expect(container).toMatchSnapshot();
    });
  });
});
