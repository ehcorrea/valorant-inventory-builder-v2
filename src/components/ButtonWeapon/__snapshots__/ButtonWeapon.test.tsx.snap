// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`given that the <ButtonWeapon> is rendered and is 'weapon' variant render image and dont render label 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "aspectRatio": 2,
      "backgroundColor": "#ffffff14",
      "borderBottomLeftRadius": 10,
      "borderBottomRightRadius": 10,
      "borderColor": "#FD4556",
      "borderStyle": "solid",
      "borderTopLeftRadius": 10,
      "borderTopRightRadius": 10,
      "borderWidth": 2,
      "justifyContent": "center",
      "marginBottom": 5,
      "marginLeft": 5,
      "marginRight": 5,
      "marginTop": 5,
      "maxHeight": 200,
      "opacity": 1,
      "overflow": "hidden",
      "width": "100%",
    }
  }
  testID="weapon-button"
>
  <View
    style={
      {
        "display": "flex",
        "height": "100%",
        "justifyContent": "center",
        "paddingBottom": 5,
        "paddingLeft": 5,
        "paddingRight": 5,
        "paddingTop": 5,
        "width": "100%",
      }
    }
    variant="weapon"
  >
    <Image
      accessibilityLabel="Weapon Display Name button"
      contentFit="contain"
      source="placeholder-image.png"
      style={
        {
          "height": "100%",
        }
      }
    />
  </View>
</View>
`;

exports[`given that the <ButtonWeapon> is rendered and is 'weapon-skin' variant  render label 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "aspectRatio": 2,
      "backgroundColor": "#ffffff14",
      "borderBottomLeftRadius": 10,
      "borderBottomRightRadius": 10,
      "borderTopLeftRadius": 10,
      "borderTopRightRadius": 10,
      "maxHeight": 200,
      "opacity": 1,
      "overflow": "hidden",
      "width": "100%",
    }
  }
  testID="weapon-button"
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FD4556",
        "height": "20%",
        "paddingHorizontal": 14,
        "width": "100%",
      }
    }
  >
    <Text
      adjustsFontSizeToFit={true}
      color="white"
      numberOfLines={1}
      style={
        {
          "color": "#FFFBF5",
          "fontFamily": "Tungsten-SemiBold",
          "fontSize": 31,
        }
      }
    >
      Weapon Display Name
    </Text>
  </View>
  <View
    style={
      {
        "display": "flex",
        "height": "80%",
        "justifyContent": "center",
        "paddingBottom": 5,
        "paddingLeft": 5,
        "paddingRight": 5,
        "paddingTop": 5,
        "width": "unset",
      }
    }
    variant="weapon-skin"
  >
    <Image
      accessibilityLabel="Weapon Display Name button"
      contentFit="contain"
      source="placeholder-image.png"
      style={
        {
          "height": "100%",
        }
      }
    />
  </View>
</View>
`;
